tkg/clusterclient/clusterclient.go

```diff
@@ -2485,36 +2483,44 @@ func (c *client) UpdateAWSCNIIngressRules(clusterName, clusterNamespace string)
 	}
 
 	cniIngressRules := awsCluster.Spec.NetworkSpec.CNI.CNIIngressRules
-	// first check if existing ingress rules contains the kapp-controller port
-	for _, ingressRule := range cniIngressRules {
-		if ingressRule.Description != "kapp-controller" {
-			continue
+	cniIngressRulesNeedUpdate := false
```

Comment:
Is there any unit test that we can add for this change ?

Reply:
yes. will add

Reply:
done. expanded the existing unit tests to cover new code